examples/SmartShop/contracts/SellerContract.aes

```diff
@@ -1,35 +1,36 @@
 payable contract Seller =
   record state = 
     { seller_address : address,
+      buyer_address : address,
       status : string,
-      contract_address : address }
+      price : int }
   
-  stateful entrypoint init() : state =
+  entrypoint init(buyer_address' : address, price' : int) : state =
     { seller_address = Call.caller,
+      buyer_address = buyer_address',
       status = "undefined",
-      contract_address = Contract.address }
+      price = price' }
 
   function check_money() : bool =
-    Contract.balance > 0
+    Contract.balance >= state.price
     
   stateful entrypoint send_item() : string =
     require(check_money(), "Not enough money")
     put(state{status = "sent_to_transport_courier"})
     state.status
+
+  function check_buyer() : bool =
+    state.seller_address != state.buyer_address
       
-  payable stateful entrypoint received_item() : bool =
+  stateful entrypoint received_item() : unit =
+    require(check_buyer(), "Buyer Only Function")
```

Comment:
So any friend of the seller can call this function? Wouldn't it make more sense if only the buyer could call it?

Reply:
Fixed in the [b182f04 ](https://github.com/aeternity/aepp-sophia-examples/pull/58/commits/b182f04aae62e33f0bdcc7793d954fd996983bd4) commit.